[new-product]  Add Chrome OS

adriens opened the pull request:
# :bookmark_tabs: Resources

Below the resources I used to fill the data:

![image](https://github.com/endoflife-date/endoflife.date/assets/5235127/81292e64-aeb4-4108-af16-d2eb25e62381)

- [Long-term support releases](https://chromeos.dev/en/education/chromeos-lts)
- [Previous release notes](https://support.google.com/chrome/a/answer/10314655)

# :pirate_flag: Scrap versions, dates and urls

Below is a [scrapper (asked by @marcwrobel )](https://github.com/endoflife-date/endoflife.date/pull/3442/files#r1306439036) to get these from python : 

Get bs4:

```shell
pip install beautifulsoup4
```

then this one does the job

```python
#!/usr/bin/env python3
import re
from datetime import datetime
import requests
from bs4 import BeautifulSoup

# utility fnctions
def extract_version_and_date(input_string):
    # Extract version number using regular expression
    version_match = re.search(r'Chrome (\d+):', input_string)
    if version_match:
        version = version_match.group(1)
    else:
        version = None

    # Extract date using datetime parsing
    # Extract date using regular expression
    date_match = re.search(r': (.+)$', input_string)
    if date_match:
        date_str = date_match.group(1)
    else:
        date_str = None

    try:
        # First, try to parse with abbreviated month name format (%b)
        try:
            date = datetime.strptime(date_str, '%b %d, %Y')
        except ValueError:
            # If that fails, try with full month name format (%B)
            date = datetime.strptime(date_str, '%B %d, %Y')
    except ValueError:
        # If that also fails, return None for both values
        date = None

    return version, date


url = "https://support.google.com/chrome/a/answer/10314655"

# Send an HTTP GET request to the URL
response = requests.get(url)

if response.status_code == 200:
    # Parse the HTML content using BeautifulSoup
    soup = BeautifulSoup(response.content, "html.parser")

    # Find the table containing Chrome OS versions
    table = soup.find("table", class_="nice-table")

    if table:
        # Find all rows in the table except the header
        rows = table.find_all("tr")[1:]

        print("Chrome OS Versions:")
        for row in rows:
            # Extract version number from the second cell of each row
            cells = row.find_all("td")
            raw_text = cells[0].get_text().strip()
            version, date = extract_version_and_date(raw_text)
            if version and date:
                print("Version:", version)
                print("Date:", date.strftime('%Y-%m-%d'))
                print("url:", "https://support.google.com/chrome/a/answer/10314655#" + version)
            else:
                print("Pattern not found in the input string.")



            #raw_text = cells[1].get_text().strip()
            #print(raw_text)
            print("######################################")

else:
    print("Failed to retrieve the web page. Status code:", response.status_code)
```

![image](https://github.com/endoflife-date/endoflife.date/assets/5235127/fdf61dba-c457-44ba-ab9d-4e5338ac67ea)
<pre><span style="background-color:#D3D7CF"><font color="#080808">   </font></span><span style="background-color:#3465A4"><font color="#D3D7CF"></font></span><span style="background-color:#3465A4"><font color="#E4E4E4">   </font></span><span style="background-color:#3465A4"><font color="#EEEEEE"><b>~</b></font></span><span style="background-color:#3465A4"><font color="#E4E4E4"> </font></span><font color="#3465A4">▓▒░</font> ./chromeos.py                                                                                                                                                                                             <font color="#2E3436">░▒▓</font><span style="background-color:#2E3436"><font color="#4E9A06"> ✔ </font></span><span style="background-color:#2E3436"><font color="#D3D7CF"></font></span><span style="background-color:#D3D7CF"><font color="#2E3436"> 19:18:50   </font></span>
Chrome OS Versions:
Version: 115
Date: 2023-07-12
url: https://support.google.com/chrome/a/answer/10314655#115
######################################
Version: 114
Date: 2023-05-24
url: https://support.google.com/chrome/a/answer/10314655#114
######################################
Version: 113
Date: 2023-04-26
url: https://support.google.com/chrome/a/answer/10314655#113
######################################
Version: 112
Date: 2023-03-29
url: https://support.google.com/chrome/a/answer/10314655#112
######################################
Version: 111
Date: 2023-03-01
url: https://support.google.com/chrome/a/answer/10314655#111
######################################
Version: 110
Date: 2023-02-01
url: https://support.google.com/chrome/a/answer/10314655#110
######################################
Version: 109
Date: 2023-01-10
url: https://support.google.com/chrome/a/answer/10314655#109
######################################
Version: 108
Date: 2022-11-29
url: https://support.google.com/chrome/a/answer/10314655#108
######################################
Version: 107
Date: 2022-10-25
url: https://support.google.com/chrome/a/answer/10314655#107
######################################
Version: 106
Date: 2022-09-27
url: https://support.google.com/chrome/a/answer/10314655#106
######################################
Version: 105
Date: 2022-08-30
url: https://support.google.com/chrome/a/answer/10314655#105
######################################
Version: 104
Date: 2022-08-02
url: https://support.google.com/chrome/a/answer/10314655#104
######################################
Version: 103
Date: 2022-06-21
url: https://support.google.com/chrome/a/answer/10314655#103
######################################
Version: 102
Date: 2022-05-26
url: https://support.google.com/chrome/a/answer/10314655#102
######################################
Version: 101
Date: 2022-04-26
url: https://support.google.com/chrome/a/answer/10314655#101
######################################
Version: 100
Date: 2022-03-29
url: https://support.google.com/chrome/a/answer/10314655#100
######################################
Version: 99
Date: 2022-03-01
url: https://support.google.com/chrome/a/answer/10314655#99
######################################
Version: 98
Date: 2022-02-02
url: https://support.google.com/chrome/a/answer/10314655#98
######################################
Version: 97
Date: 2022-01-04
url: https://support.google.com/chrome/a/answer/10314655#97
######################################
Version: 96
Date: 2021-12-06
url: https://support.google.com/chrome/a/answer/10314655#96
######################################
Version: 95
Date: 2021-10-19
url: https://support.google.com/chrome/a/answer/10314655#95
######################################
Pattern not found in the input string.
######################################
Version: 94
Date: 2021-09-21
url: https://support.google.com/chrome/a/answer/10314655#94
######################################
Version: 93
Date: 2021-08-31
url: https://support.google.com/chrome/a/answer/10314655#93
######################################
Version: 92
Date: 2021-07-20
url: https://support.google.com/chrome/a/answer/10314655#92
######################################
Version: 91
Date: 2021-05-25
url: https://support.google.com/chrome/a/answer/10314655#91
######################################
Version: 90
Date: 2021-04-14
url: https://support.google.com/chrome/a/answer/10314655#90
######################################
Version: 89
Date: 2021-03-09
url: https://support.google.com/chrome/a/answer/10314655#89
######################################
Version: 88
Date: 2021-01-19
url: https://support.google.com/chrome/a/answer/10314655#88
######################################
Version: 87
Date: 2020-11-17
url: https://support.google.com/chrome/a/answer/10314655#87
######################################
Version: 86
Date: 2020-10-06
url: https://support.google.com/chrome/a/answer/10314655#86
######################################
Version: 85
Date: 2020-08-25
url: https://support.google.com/chrome/a/answer/10314655#85
######################################
</pre>

adriens committed:
Create chromeos.md

hebbet committed:
Update chromeos.md

hebbet committed:
Update chromeos.md

hebbet committed:
Update chromeos.md

marcwrobel commented on products/quarkus.md:
```diff
@@ -7,7 +7,7 @@ permalink: /quarkus-framework
 -   /quarkus
 releasePolicyLink: https://quarkus.io/security/
-changelogTemplate: "https://github.com/quarkusio/quarkus/releases/tag/__LATEST__.Final"
+changelogTemplate: "https://github.com/quarkusio/quarkus/releases/tag/__LATEST__"
```
This change is not related to Chrome OS and must be reverted.

FYI this Quarkus change has been taken into account in #3467.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,150 @@
+-   /chrome-os
+versionCommand: cat /etc/issue
+releasePolicyLink: https://chromeos.dev/en/education/chromeos-lts
+changelogTemplate: https://support.google.com/chrome/a/answer/10314655/__RELEASE_CYCLE__
```
```suggestion
changelogTemplate: https://support.google.com/chrome/a/answer/10314655#__RELEASE_CYCLE__
```

It's does not look like it is working with a '/'. Links in the `Chrome version & Stable channel release date` are using a dash.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,150 @@
+
+---
+
+> [chromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source ChromiumOS and uses the Google Chrome web browser as its principal user interface.
```
The description should summarize the release / EOL / LTS policy.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,150 @@
+activeSupportColumn: false
+releaseDateColumn: true
+
+releases:
```
Would be great to have a custom script scraping information in the _Chrome version & Stable channel release date_ on https://support.google.com/chrome/a/answer/10314655.

adriens committed:
Update products/chromeos.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

adriens committed:
Update chromeos.md

As of https://github.com/endoflife-date/endoflife.date/pull/3442#discussion_r1306438933

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,151 @@
+---
+> Chrome OS follows a regular release schedule, similar to other operating systems. New versions of Chrome OS are released approximately every six weeks. These releases include updates, bug fixes, security enhancements, and sometimes new features. Chrome OS releases are identified by version numbers, such as "Chrome OS 95," "Chrome OS 96," and so on.
+> Devices receive automatic software updates for approximately 6.5 years from their original launch date. During the first 5 years, devices receive both security and feature updates. In the last 1.5 years, devices continue to receive security updates but might not receive feature updates.
+> [chromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source ChromiumOS and uses the Google Chrome web browser as its principal user interface.
```
Looks like the product description and the release policy has been mixed up.

marcwrobel replied:
@adriens, this has not been fixed in your latest commit.

marcwrobel commented on products/quarkus.md:
```diff
@@ -7,7 +7,8 @@ permalink: /quarkus-framework
 -   /quarkus
 releasePolicyLink: https://quarkus.io/security/
-changelogTemplate: "https://github.com/quarkusio/quarkus/releases/tag/__LATEST__.Final"
+# See https://github.com/endoflife-date/endoflife.date/issues/3462
```
Please remove any changes related to Quarkus from this PR.

marcwrobel reviewed (changes requested):
Not good: this PR remove the quarkus.md file entirely.

adriens committed:
Update chromeos.md

As of https://github.com/endoflife-date/endoflife.date/pull/3442#discussion_r1327985155

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+releases:
+
+-   releaseCycle: "116"
+    releaseDate: 2023-08-22
```
2023-08-09 according to https://support.google.com/chrome/a/answer/10314655.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+
+-   releaseCycle: "85"
+    releaseDate: 2020-08-25
+    eol: 2020-10-06
```
As per https://blog.chromium.org/2021/06/changes-to-chrome-oss-release-cycle.html, I think the information taken from https://support.google.com/chrome/a/answer/10314655 is only valid from Chrome 96 onward:

- chrome OS 94 has its own entry
- chrome OS 95 was skipped according to https://blog.chromium.org/2021/06/changes-to-chrome-oss-release-cycle.html

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+    eol: 2020-10-06
+
+---
+> [chromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source ChromiumOS and uses the Google Chrome web browser as its principal user interface.
```
```suggestion
> [ChromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source Chromium OS and uses the Google Chrome web browser as its principal user interface.
```

According to https://www.chromium.org/chromium-os/.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+---
+> [chromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source ChromiumOS and uses the Google Chrome web browser as its principal user interface.
+
+Chrome OS follows a regular release schedule, similar to other operating systems. New versions of Chrome OS are released approximately every six weeks. These releases include updates, bug fixes, security enhancements, and sometimes new features. Chrome OS releases are identified by version numbers, such as "Chrome OS 95," "Chrome OS 96," and so on.
```
4 weeks according to https://blog.chromium.org/2021/06/changes-to-chrome-oss-release-cycle.html. https://chromeos.dev/en/education/chromeos-lts is even more precise on the subject and maybe it should be taken as the reference ?

The sentence "Chrome OS releases are identified by version numbers, such as "Chrome OS 95," "Chrome OS 96," and so on." looks superfluous as the table with versions is displayed above.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+    releaseDate: 2023-01-10
+    eol: 2023-02-01
+
+-   releaseCycle: "108"
```
[LTS releases](https://chromeos.dev/en/education/chromeos-lts) are not marked as such.

marcwrobel replied:
LTS releases still need to be marked as such.

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+alternate_urls:
+-   /chrome-os
+versionCommand: cat /etc/issue
+releasePolicyLink: https://chromeos.dev/en/education/chromeos-lts
```
I'm a bit confused by the "Education" label. Is what's documented on https://chromeos.dev/en/education/chromeos-lts valid for all Chrome OS versions? Is there a similar page that does not labelled "Education" ?

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,153 @@
+
+Chrome OS follows a regular release schedule, similar to other operating systems. New versions of Chrome OS are released approximately every six weeks. These releases include updates, bug fixes, security enhancements, and sometimes new features. Chrome OS releases are identified by version numbers, such as "Chrome OS 95," "Chrome OS 96," and so on.
+
+Devices receive automatic software updates for approximately 6.5 years from their original launch date. During the first 5 years, devices receive both security and feature updates. In the last 1.5 years, devices continue to receive security updates but might not receive feature updates.
```
This needs to be sourced. Where can I confirmed this (would prefer an official "Google" link if possible).

adriens committed:
Update products/chromeos.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

adriens committed:
Update chromeos.md

As of https://github.com/endoflife-date/endoflife.date/pull/3442#discussion_r1329264349

adriens committed:
Update chromeos.md

As of https://github.com/endoflife-date/endoflife.date/pull/3442#discussion_r1329272630

adriens committed:
Update chromeos.md

as of https://github.com/endoflife-date/endoflife.date/pull/3442#discussion_r1329281011

marcwrobel commented on products/chromeos.md:
```diff
@@ -0,0 +1,150 @@
+---
+> [ChromeOS](https://www.google.com/chromebook/chrome-os/) is a Linux-based operating system developed and designed by Google. It is derived from the open-source Chromium OS and uses the Google Chrome web browser as its principal user interface.
+
+According to [Auto Update policy](https://support.google.com/chrome/a/answer/6220366?hl=en), Chrome devices (e.g. Chromebook, Chromebox, Chromebase, Chromebit) receive automatic updates that enhance both the device and its software. Automatic updates provide the latest features, keep the device secure, and are applied across the operating system, browser and hardware. These updates depend on many ...
```
This description is about devices, not ChromeOS per se.

What's written on the previous page you used looks more focus on ChromeOS: https://chromeos.dev/en/education/chromeos-lts and provide a little bit more insights:

- stable : 1 release every 4 weeks
- LTS : 1 per 6 months

It does not say how much time stable and LTS releases are supported though. Maybe a stable is supported until the next, and similarly an LTS is supported until the next LTS ?
